### ISSUE TITLE
Block backgrounds don't work.

### ISSUE THREAD

**Issue by transientlunatic** [transientlunatic/glasgow-theme#2]
To quote @seands:

> I'm trying to make block backgrounds appear in a beamer document using
> your Glasgow beamer theme. I noticed in the source that there's an
> option 'blockbg' that I can pass, which seems to change colours. I tried
> adding it to the preamble, i.e.
> 
> \usetheme[blockbg]{glasgow}
> 
> But that gives me an error:
> 
> ! Package xcolor Error: Undefined color `palette primary.bg'.
> 
> See the xcolor package documentation for explanation.
> Type  H <return>  for immediate help.
>  ...
> 
> l.92 \end{frame}
> 
> Is this supposed to work? I noticed that there are no colours defined as
> 'palette primary.bg', etc., only 'palette primary'. Also, ideally I'd
> like to colour only one block's background - is that possible?>

**Comment by SeanDS** [comment]
Minimum non-working example:

```latex
\documentclass[10pt, compress]{beamer}

\usetheme[blockbg]{glasgow}

\begin{document}

\begin{frame}
  \frametitle{Frame}
  \begin{block}{Block}
    Contents
  \end{block}
\end{frame}

\end{document}
```

Of course, make sure you have `UoG_keyline.pdf`, `beamercolorthemeglasgow.sty`, `beamerfontthemeglasgow.sty` and `beamerthemeglasgow.sty` in the same directory.

I'm using `texlive-full 2016.20160819-2` on Ubuntu 16.10.

**Comment by transientlunatic** [comment]
This is a little odd, but I think the problem is because you don't have a `\section{}` macro before the block. In the meantime a workaround is to include on e.g.

```latex
\documentclass[10pt, compress]{beamer}

\usetheme[blockbg]{glasgow}

\begin{document}
\section{Section A}
\begin{frame}
  \frametitle{Frame}
  \begin{block}{Block}
    Contents
  \end{block}
\end{frame}

\end{document}
```
works correctly for me, but it seems sub-optimal that you would /need/ to do this.

**Comment by SeanDS** [comment]
Hah! That's strange. Yeah, I'd like to be able to avoid having section slides but still use blocks with solid backgrounds.

**Comment by transientlunatic** [comment]
Ahah! Found the problem, and fixed it in commit 1d2fffe.
